### PR TITLE
update autolaoder usage

### PR DESCRIPTION
**Pull request by azjezz**

### CODE REVIEW

File: bin/hacktest [hunk 1]
```diff
@@ -1,40 +1,2 @@
...
-  fprintf(STDERR, "Failed to find autoloader.\n");
-  exit(1);
-}
-
-<<__Entrypoint>>
-async function hackTestMainAsync(): Awaitable<noreturn> {
-  $exit_code = await HackTestCLI::runAsync();
-  exit($exit_code);
-}
+#!/bin/sh
```
Comment (fredemmott) [inline bin/hacktest]: the 755 -> 644 mode change looks incorrect

File: bin/hacktest [hunk 2]
```diff
@@ -1,40 +1,2 @@
...
+exec "$(dirname $0)/hacktest.hack" "$@"
```
Comment (fredemmott) [inline bin/hacktest]: Whoops, I messed this up in hh-autoload - `$0` needs to be double-quoted too, within the subshell

File: bin/hacktest.hack [hunk 3]
```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env hhvm
```
Comment (fredemmott) [inline bin/hacktest.hack]: What mode is this?

File: bin/hacktest.hack [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+<<__EntryPoint>>
+async function hackTestMainAsync(): Awaitable<noreturn> {
+  $root = \realpath(__DIR__.'/..');
+  $found_autoloader = false;
+  while (true) {
+    $autoloader = $root.'/vendor/autoload.hack';
+    if (\file_exists($autoloader)) {
+      $found_autoloader = true;
+      require($autoloader);
```
Comment (fredemmott) [inline bin/hacktest.hack]: No typechecker error for this?

Comment (azjezz) [reply bin/hacktest.hack]: nope.

File: bin/hacktest (hunk 2 above)
Comment (azjezz) [reply bin/hacktest]: it works fine ?
actually, when i changed it to `exec "$(dirname \"$0\")/hacktest.hack" "$@"`, it didn't work :confused:

File: bin/hacktest.hack (hunk 3 above)
Comment (azjezz) [reply bin/hacktest.hack]: file mode ? 
```
775 hacktest
755 hacktest.hack
```

File: bin/hacktest (hunk 2 above)
Comment (fredemmott) [reply bin/hacktest]: No backticks.

Problem is:
- clone to a directory containing a space in the name
- go out of that directory (e.g. `cd /tmp`)
- `/path/including a space/hacktest/bin/hacktest`

File: bin/hacktest.hack (hunk 4 above)
Comment (azjezz) [reply bin/hacktest.hack]: the only type check error in there is for `Facebook\AutoloadMap\initialize()` ( Unbound name ) :stuck_out_tongue:

File: bin/hacktest (hunk 2 above)
Comment (azjezz) [reply bin/hacktest]: fix.

**Comment by fredemmott** [comment]
Thanks - this was effectively part of 3ad1281

**Pull request closed without merging**